specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1082 @@
+    // ID of the current network
+    NetworkName     address.SubnetID      
+    // Human-readable name for the subnet
+    Name            address.SubnetID  
+    // Consensus implemented in the subnet
+    Consensus hierarchical.ConsensusType
+    // Arbitrary consensus parameters required to initialize
+    // the consensus protocol (e.g. minimum number of validators).
+    // These parameters are consensus-specific.
+    ConsensusParams *hierarchical.ConsensusParams
+    // Minimum amount of stake required from participant 
+    // to become a validator of the subnet.
+    MinValidatorStake   abi.TokenAmount
+    // Checkpointing period used in the subnet. It determines
+    // how often the subnet propagates and commits checkpoints in its
+    // parent
+    CheckPeriod     abi.ChainEpoch
+}
+```
+> TODO: There is an on-going revision of `ConstructParams` to include a way to provide arbitrary input arguments to a subnet.
+
+## Subnet Coordinator Actor (SCA)
+The main entity responsible for handling all the lifecycle of child subnets in a specific chain is the Subnet Coordinator Actor (`SCA`). The `SCA` is a builtin-actor that exposes the interface for subnets to interact with the hierarchical consensus protocol. This actor includes all the available functionalities related to subnets and their management. It also enforces all the security requirements, fund management, and cryptoeconomics of the hierarchical consensus, as Subnet Actors are user-defined and can't be (fully) trusted. The `SCA` has a reserved address ID `f064`.
+
+The MVP implementation of this builtin-actor can be found [here](https://github.com/adlrocha/builtin-actors/tree/master/actors/hierarchical_sca). The `SCA` exposes the following functions:
+
+
+##### SCA Functions
+> Functions of the SCA. These functions are triggered when a `message` is sent for their corresponding `methodNum`.
+```go
+type SCA interface{
+    // Initializes the state of the SCA and sets
+    // all its initial parameters.
+    // (methodNum = 1)
+    Constructor(ConstructParams)
+    
+    // Register expects as source the address of the subnet actor of
+    // the subnet that wants to be registered. Its
+    // `value` an amount of collateral over the `CollateralThreshold`.
+    // This functions activates the subnet and from there on other
+    // subnets in the system are allowed to interact with it and the
+    // subnet can start commtting its checkpoints.
+    // (methodNum = 2)
+    Register()
+    
+    // AddCollateral expects as source the address of the subnet actor
+    // for which the collateral wants to be added.  Its `value` should
+    // include the amount of collateral to be added for the subnet.
+    // (methodNum = 3)
+    AddCollateral()
+    
+    // ReleaseCollateral expects as source the address of the subnet actor
+    // for which the collateral should be released. It triggers a message
+    // to the subnet actor returning the corresponding collateral. 
```

Comment:
Do we know which message? Or is this just some "transfer" type? In the other direction it says specifically what type of message is sent, that's why I'm asking.

Reply:
Yes, it's a transfer message. Adding a clarification.